tests/utils/test_alignment_utils.py

```diff
@@ -0,0 +1,864 @@
+        computed_point3s = [x.translation() for x in computed]
+        expected_rot3s = [x.rotation() for x in expected]
+        expected_point3s = [x.translation() for x in expected]
+
+        self.__assert_equality_on_rot3s(computed_rot3s, expected_rot3s)
+        self.__assert_equality_on_point3s(computed_point3s, expected_point3s)
+
+    def setUp(self):
+        super().setUp()
+
+        self.addTypeEqualityFunc(Rot3, rot3_compare)
+        self.addTypeEqualityFunc(Point3, point3_compare)
+
+    def test_align_rotations(self):
+        """Tests the alignment of rotations."""
+
+        # using rotation along just the Y-axis so that angles can be linearly added.
+        input_list = [
+            Rot3.RzRyRx(np.deg2rad(0), np.deg2rad(-10), 0),
+            Rot3.RzRyRx(np.deg2rad(0), np.deg2rad(30), 0),
+        ]
+        ref_list = [
+            Rot3.RzRyRx(np.deg2rad(0), np.deg2rad(80), 0),
+            Rot3.RzRyRx(np.deg2rad(0), np.deg2rad(-40), 0),
+        ]
+
+        computed = alignment_utils.align_rotations(input_list, ref_list)
+        expected = [
+            Rot3.RzRyRx(0, np.deg2rad(80), 0),
+            Rot3.RzRyRx(0, np.deg2rad(120), 0),
+        ]
+
+        self.__assert_equality_on_rot3s(computed, expected)
+
+    def test_align_rotations_with_no_relative_rotations(self):
+        """Tests the alignment of rotations."""
+
+        # using rotation along just the Y-axis so that angles can be linearly added.
+        input_list = [
+            Rot3.RzRyRx(np.deg2rad(0), np.deg2rad(-10), 0),
+            Rot3.RzRyRx(np.deg2rad(0), np.deg2rad(30), 0),
+        ]
+        ref_list = copy.deepcopy(input_list)
+
+        computed = alignment_utils.align_rotations(input_list, ref_list)
+        expected = copy.deepcopy(input_list)
+        self.__assert_equality_on_rot3s(computed, expected)
+
+    def test_align_poses_after_sim3_transform(self):
+        """Test for alignment of poses after applying a SIM3 transformation."""
+
+        translation_shift = np.array([5, 10, -5])
+        rotation_shift = Rot3.RzRyRx(0, 0, np.deg2rad(30))
+        scaling_factor = 0.7
+
+        transform = Similarity3(rotation_shift, translation_shift, scaling_factor)
+        ref_list = [transform.transformFrom(x) for x in sample_poses.CIRCLE_TWO_EDGES_GLOBAL_POSES]
+
+        computed_poses, aSb = alignment_utils.align_poses_sim3(sample_poses.CIRCLE_TWO_EDGES_GLOBAL_POSES, ref_list)
+        assert isinstance(aSb, Similarity3)
+        self.__assert_equality_on_pose3s(computed_poses, sample_poses.CIRCLE_TWO_EDGES_GLOBAL_POSES)
+
+    def test_align_poses_with_outlier(self):
+        """Test for alignment of poses after applying a SIM3 transform and adding an outlier."""
+        translation_shift = np.array([5, 10, -5])
+        rotation_shift = Rot3.RzRyRx(0, 0, np.deg2rad(30))
+        scaling_factor = 0.7
+
+        transform = Similarity3(rotation_shift, translation_shift, scaling_factor)
```

Comment:
can you rename this variable to `aSb`? Otherwise sounds like the name of a python module